nonlinear_elasticity/nonlinear_elasticity.cc

```diff
@@ -1501,11 +1502,29 @@ main(int argc, char **argv)
   using namespace Nonlinear_Elasticity;
   using namespace dealii;
 
-  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv);
 
   try
     {
       deallog.depth_console(0);
+      const int n_threads = dealii::MultithreadInfo::n_threads();
+
+      std::cout
+        << "-----------------------------------------------------------------------------"
+        << std::endl
+        << "--     . running with " << n_threads << " thread"
+        << (n_threads == 1 ? "" : "s") << std::endl;
+
+      std::cout << "--     . revision " << GIT_SHORTREV << " on branch "
+                << GIT_BRANCH << std::endl;
+      std::cout << "--     . deal.II " << DEAL_II_PACKAGE_VERSION
+                << " (revision " << DEAL_II_GIT_SHORTREV << " on branch "
+                << DEAL_II_GIT_BRANCH << ")" << std::endl;
```

Comment:
I guess this is the standard way how do that in deal.II? What happens if you didn't clone deal.II (but just installed the sources etc.)?

Reply:
We also have something similar in the preCICE CMakeLists.txt, it checks if the project comes from Git first. Maybe this helps.

Reply:
In case of the DEAL_II flags, the information is directly compiled into the library, which is always associated to some revision. I don't think we get any problem. 

In case of our own repo, the `.git` directory is sourced explicitly and when it is not found, an empty string is 'printed'. However, cloning the adapter is more or less the standard way.

Reply:
But also e.g. downloading a zip of the latest state works, right? This is something that preCICE users often do.

Reply:
Let me test, I guess no. Otherwise I need to set some default values.

Reply:
Worst case, we receive sth like
```
-----------------------------------------------------------------------------
--     . running with 4 threads
--     . revision  on branch 
--     . deal.II 9.2.0 (revision  on branch )
-----------------------------------------------------------------------------
```

Reply:
In the end it should be readable, so `not git repo found` or similar is better than just an empty string. The branch we cannot really say. You could also just leave out that part.
```
-----------------------------------------------------------------------------
--     . running with 4 threads
--     . unknown adapter revision 
--     . deal.II 9.2.0
-----------------------------------------------------------------------------
```